src/plugins/explore/public/components/visualizations/utils/utils.ts

```diff
@@ -129,3 +140,86 @@ export function swapAxes(axes: StandardAxes[]) {
     return axis;
   });
 }
+
+const positionSwapMap: Record<Positions, Positions> = {
+  [Positions.LEFT]: Positions.BOTTOM,
+  [Positions.RIGHT]: Positions.TOP,
+  [Positions.BOTTOM]: Positions.LEFT,
+  [Positions.TOP]: Positions.RIGHT,
+};
+
+const swapPosition = (pos: Positions): Positions => positionSwapMap[pos] ?? pos;
+
+export function getSwappedAxes(
+  xAxis: CompleteAxisWithStyle,
+  yAxis: CompleteAxisWithStyle,
+  switchAxes: boolean
+): [CompleteAxisWithStyle, CompleteAxisWithStyle] {
+  if (!switchAxes) {
+    return [xAxis, yAxis];
+  }
+
+  // Swap the positions, update the position field
+  // merge styles with axis mapping
+  const swappedXAxis = {
+    ...xAxis,
+    styles: {
+      ...xAxis.styles,
+      position: swapPosition(xAxis.styles.position),
+    },
+  };
+
+  const swappedYAxis = {
+    ...yAxis,
+    styles: {
+      ...yAxis.styles,
+      position: swapPosition(yAxis.styles.position),
+    },
+  };
+
+  // Swap X and Y axis data
+  return [swappedYAxis, swappedXAxis];
+}
+
+export const getSwappedAxisRole = (
+  styles: Partial<AxisSupportedStyles>,
+  axisColumnMappings?: AxisColumnMappings
+): [CompleteAxisWithStyle | undefined, CompleteAxisWithStyle | undefined] => {
+  const xMapping = axisColumnMappings?.x;
+  const yMapping = axisColumnMappings?.y;
+
+  const xAxis = getAxisByRole(styles.StandardAxes ?? [], AxisRole.X);
+  const yAxis = getAxisByRole(styles.StandardAxes ?? [], AxisRole.Y);
+
+  if (!xAxis || !yAxis) {
+    return [undefined, undefined];
+  }
+
+  const xAxisWithMapping: CompleteAxisWithStyle = {
+    ...xMapping,
+    styles: xAxis,
+  };
+
+  const yAxisWithMapping: CompleteAxisWithStyle = {
+    ...yMapping,
+    styles: yAxis,
+  };
+
+  // Swap axes and their positions based on switchAxes flag
+  return getSwappedAxes(xAxisWithMapping, yAxisWithMapping, styles?.switchAxes ?? false);
```

Comment:
I feel it's unnecessary to create another type `CompleteAxisWithStyle`, but we can refactor this after this PR, can you add a TODO for now?
```suggestion
  if (!styles?.switchAxes ) {
    return [xAxis, xMapping, yAxis, yMapping];
  }

  // Swap axes and their positions based on switchAxes flag
  return getSwappedAxes(xAxis, xMapping, yAxis, yMapping);
```